src/qtism/data/storage/xml/XmlDocument.php

```diff
@@ -75,6 +77,21 @@ class XmlDocument extends QtiDocument
      */
     private $filesystem;
 
+    public function __serialize(): array
+    {
+        return ['domDocument' => (string)$this];
+    }
+
+    public function __unserialize(array $data): void
+    {
+        $this->loadFromString($data['domDocument']);
+    }
+
+    public function __toString(): string
+    {
+        return $this->saveToString(false);
```

Comment:
Is it ok that now we could have exception on serialization if no QtiComponent was inside ?

Because that situation could be 
https://github.com/oat-sa/qti-sdk/blob/28cd6bd951161afeab9f2273e82103ca392b6390/src/qtism/data/QtiDocument.php#L60

Reply:
The only way to initialize data in `XmlDocument` is to call one of the `load` methods. That will set the component as well.

If no data was loaded, I presume there was no reason to serialize that object.